### PR TITLE
lesson18-assignment

### PR DESCRIPTION
**Pull request by rdnnwe**
Test PR

### CODE REVIEW

File: vpc.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module "vpc" {
+    source = "github.com/terraform-aws-modules/terraform-aws-vpc"
```
Comment (vladibb) [inline vpc.tf:2]: remove github.com (since it's auto expanding github)

File: vpc.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    
+    name = "test-vpc"
+    cidr = "10.1.0.0/16"
+    azs  = ["eu-west-1a", "eu-west-1b"]
+    private_subnets = ["10.1.0.0/18", "10.1.128.0/18"]
+    public_subnets  = ["10.1.64.0/18", "10.1.192.0/18"]
```
Comment (vladibb) [inline vpc.tf:8]: all cidr values shall be moved into locals { } section, refactor plz

**Review by vladibb — CHANGES_REQUESTED** [review]

File: aws-lb-target-group.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+resource "aws_lb_target_group_attachment" "test" {
+    target_group_arn = module.alb.target_group_arns[0]
+    target_id        = aws_instance.RDN_Test_TerraformInstance_01.private_ip
+    port             = 80
+}
```
Comment (vladibb) [inline aws-lb-target-group.tf:5]: refactor into alb.tf - this belongs to alb

File: main.tf [hunk 4]
```diff
@@ -1,106 +1,6 @@
...
-            type    = "A"
-            alias   = {
-                name    = module.alb.this_lb_dns_name
-                zone_id = module.alb.this_lb_zone_id
-            }
-        }
-    ]
+  profile = "default"
+  region = "var.region"
 }
```
Comment (vladibb) [inline main.tf:4]: this is aws.tf - refactor into the new file

File: records.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
+module "records" {
+    source  = "terraform-aws-modules/route53/aws//modules/records"
+    zone_id = "Z05830172NQEHW4JNM4HI"
```
Comment (vladibb) [inline records.tf:3]: static value, refactor into local, name variable **public_zone_id**

File: vpc.tf [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    
+    enable_nat_gateway = true
+    enable_dns_support = true
+
+    tags = {
+        Terraform = "true"
+        Environment = "dev"
+    }
+}
```
Comment (vladibb) [inline vpc.tf:17]: refactor into main.tf locals, map object **common_tags** see 18.6 from home assignment todo

File: vpc.tf (hunk 1 above)
Comment (rdnnwe) [reply vpc.tf:2]: Done

File: aws-lb-target-group.tf (hunk 3 above)
Comment (rdnnwe) [reply aws-lb-target-group.tf:5]: Explain please

File: vpc.tf (hunk 2 above)
Comment (rdnnwe) [reply vpc.tf:8]: Explain, please

File: aws-lb-target-group.tf (hunk 3 above)
Comment (vladibb) [reply aws-lb-target-group.tf:5]: alb.tf + aws-lb-target-group.tf = alb.tf

**Pull request merged**